ballerina-shell/modules/shell-cli/src/test/java/io/ballerina/shell/cli/test/base/TestIntegrator.java

```diff
@@ -57,61 +62,84 @@ public TestIntegrator(InputStream inputStream, OutputStream outputStream,
     @Override
     public void run() {
         try {
-            String shellPrompt = PropertiesLoader.getProperty(REPL_PROMPT);
             PrintStream testPrint = new PrintStream(outputStream, true, Charset.defaultCharset());
             InputStreamReader inStreamReader = new InputStreamReader(inputStream, Charset.defaultCharset());
             BufferedReader testReader = new BufferedReader(inStreamReader);
 
+            sendRequest(testPrint, "");
+            readResponse(testReader);
+            readResponse(testReader);
```

Comment:
Why are we reading twice here? And the read response also ignored?

Reply:
The response here is not testable because they can change. 
The first `readResponse` is to read and ignore all the initial output from the CLI. (header text, etc...)
The second `readResponse` is to read any response to the first request. (L69)

After that we can send a request and read its correct response.

Reply:
Ok, then can you add a comment describing this?